### PR TITLE
Move to Jenkinsfile for builds

### PR DESCRIPTION
**Pull request by sgtcoolguy**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,104 @@
+@NonCPS
+def jsonParse(def json) {
+	new groovy.json.JsonSlurperClassic().parseText(json)
+}
+
+def nodeVersion = '4.7.3'
+def tiSDKVersion = '5.3.0.GA'
```
Comment (hansemannn) [inline Jenkinsfile]: We should use 6.0.2.GA

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            // if lib folder doesn't exist, create it
+            sh 'mkdir -p lib'
+            // if build folder doesn't exist, create it
+            sh 'mkdir -p build'
+            // if build/docs folder doesn't exist, create it
+            sh 'mkdir -p build/docs'
+            def antHome = tool(name: 'Ant 1.9.2', type: 'ant')
+            withEnv(["PATH+ANT=${antHome}/bin"]) {
+              sh 'ant clean'
+              sh 'ant'
```
Comment (hansemannn) [inline Jenkinsfile]: We should use `ti build -p android --build-only` or the Appc-CLI `appc run -p android --build-only`.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        dir('ios') {
+          writeFile file: 'titanium.xcconfig', text: """
+TITANIUM_SDK = ${activeSDKPath}
+TITANIUM_BASE_SDK = \"\$(TITANIUM_SDK)/iphone/include\"
+TITANIUM_BASE_SDK2 = \"\$(TITANIUM_SDK)/iphone/include/TiCore\"
+TITANIUM_BASE_SDK3 = \"\$(TITANIUM_SDK)/iphone/include/ASI\"
+TITANIUM_BASE_SDK4 = \"\$(TITANIUM_SDK)/iphone/include/APSHTTPClient\"
+HEADER_SEARCH_PATHS= \$(TITANIUM_BASE_SDK) \$(TITANIUM_BASE_SDK2) \$(TITANIUM_BASE_SDK3) \$(TITANIUM_BASE_SDK4) \${PROJECT_DIR}/**
+"""
+          sh './build.py'
```
Comment (hansemannn) [inline Jenkinsfile]: We should use `ti build -p ios --build-only` or the Appc-CLI `appc run -p ios --build-only`.

**Review by hansemannn — CHANGES_REQUESTED** [review]

**Review by hansemannn — APPROVED** [review]

**Comment by sgtcoolguy** [comment]
https://jira.appcelerator.org/browse/TIMOB-24480 is breaking the `appc ti build` for android here. Not sure if we want to just get that merged and update the Jenkinsfile here to use 6.0.3.GA whenever that comes out, or use ant to build the android half for now and update when 6.0.3.GA goes out.

**Pull request merged**